### DISCUSSION TITLE
How can I set int value from radio buttons

### DISCUSSION THREAD

**Discussion by nani1008** [react-hook-form/react-hook-form#6607] [Q&A] [answered]
I want to set my value from my multiple radio buttons with int type but it is returning string.

tried putting parseInt on the value still doesnt work.

`<input value={1} type=“radio” {...register("number")} /> `

**Comment by EverStarck** [comment, marked as answer]
Did you try with `defaultValues`?

```
const { register } = useForm({
  defaultValues: {
    firstName: "bill",
    lastName: "luo",
  }
})

<input {...register("firstName")} /> // ✅ working version
<Controller name="lastName" render={({ field }) => <input {...field} />} />  // ✅ working version
```
[Here you have the doc](https://react-hook-form.com/api/useform#useFormRef)
[Here are an example with radio](https://codesandbox.io/s/react-hook-form-defaultvalues-v7-vd85w?file=/src/index.js:298-315)

**Reply by nani1008** [reply]
@EverStarck If I have 1 input field I can set the value to int but if I have multiple it says “on”

**Reply by EverStarck** [reply]
Do you have a csb?
